File: tests/workflow/consumer/rust/src/consumer/broadcasts/groupb_groupc_structa.rs
use super::{protocol, Consumer, Context};
use clibri::client;

pub async fn handler<E: client::Error>(
    event: protocol::GroupB::GroupC::StructA,
    context: &mut Context,
    consumer: Consumer<E>,
) {
    context.broadcast.groupb_groupc_structa += 1;
    context.broadcast.check();
}

File: experimental/workflow_mt/producer/typescript/src/producer/beacons/groupb.groupc.structa.ts
import {
    Context,
    Producer,
    Identification,
    Filter,
    Protocol,
} from "../implementation/beacons";

export function emit(
    beacon: Protocol.GroupB.GroupC.StructA,
    consumer: Identification,
    filter: Filter,
    context: Context,
    producer: Producer
): Promise<void> {
    return Promise.reject(
        new Error(`Handler for Protocol.GroupB.GroupC.StructA isn't implemented.`)
    );
}